sdk/cdsclient/client.go

```diff
@@ -48,7 +48,7 @@ func NewService(endpoint string, timeout time.Duration) Interface {
 }
 
 // NewWorker returns client for a worker
-func NewWorker(endpoint string, name string) Interface {
+func NewWorker(endpoint string, name string, insecureSkipVerifyTLS bool) Interface {
```

Comment:
i would prefer to able to pass an optionnal *http.Client to the NewWorker funct

Reply:
I can change the code for a single client that overrides the two others (SDK and worker's client) if not nil. By the way there is currently a 10 seconds timeout for the worker's client but nothing for the SDK, should it really be the same client ?
Another solution would be to add the SDK override in a cmd prerun for worker like it is for CDS CLI and let the insecure flag as in NewHatchery func just below NewWorker.

Reply:
We want to remove cobra (and viper) from the worker, so don't consider the second solution :)
You are facing to the fact that we are currently in refactoring the code from the old sdk package and the new cdsclient package, so don't waste your time with the sdk package :)